### DISCUSSION TITLE
Auto formatting

### DISCUSSION THREAD

**Discussion by JacobBumgarner** [REditorSupport/vscode-R#1203] [Ideas]
Does vscode-R have some functionality that enables autoformatting of code with a shortcut?

I'm imagining something similar to what was described in this post:
https://stackoverflow.com/a/34856235/15481857

**Comment by renkun-ken** [comment]
You could bind shortcut keys to the following in "Keyboard shortcuts":

```json
{
  "key": "shift+alt+f",
  "command": "editor.action.formatDocument",
  "when": "editorHasDocumentFormattingProvider && editorTextFocus && !editorReadonly && !inCompositeEditor"
},
{
  "key": "cmd+k cmd+f",
  "command": "editor.action.formatSelection",
  "when": "editorHasDocumentSelectionFormattingProvider && editorTextFocus && !editorReadonly"
}
```

**Reply by werkstattcodes** [reply]
Apologies for intruding your conversation - I was also looking for a formatting shortcut and inserted the code above. As a response I get that no formatter is installed; when trying to install a formatter, the reply is that there is no formatter for quarto (which I am using with R). Is there something I can do about it? Would imagine many VS Code & R Users will also use quarto. Many thanks.

**Reply by renkun-ken** [reply]
Maybe https://github.com/quarto-dev/quarto-vscode/issues is a better place to ask? Not sure if how formatters come into play in a document with embedded languages provided by external language servers.

**Reply by JacobBumgarner** [reply]
Thanks for this, @renkun-ken.

I wonder if this could be highlighted in the wiki?

**Reply by JacobBumgarner** [reply]
And FYI for future mac users:

"option + shift + F" is already bound to "Format document" by default.
